### DISCUSSION TITLE
How many packages in AL2023 vs EPEL?

### DISCUSSION THREAD

**Discussion by mimmus** [amazonlinux/amazon-linux-2022#798] [General]
Hi,
we should compare the number of packages in AL2023 repo compared to those available in EPEL, to evaluate this operating system in our environment.
The concern is that we will not find the packages needed for our needs.

**Comment by elsaco** [comment]
`EPEL` is Extra Packages for Enterprise Linux, specifically RHEL. What the community would benefit from is  a `EPAL` repo (Extra Packages for Amazon Linux). EPEL is way out of sync with AL2023.

**Comment by danie-dejager** [comment]
Every time a package is requested to be included to AL2023 the community is asking Amazon to support it too. Every time. Red Hat does not even do that for RHEL.
I agree with @elsaco, there must be a community driven repo with packages that is maintained by the community. Amazon and the community setting the rules to what can be added to the repo. Maybe in future versions Amazon can choose the same Fedora version that became CentOS 10 Stream for AL2025 to make it easier to build existing EPEL packages. Some packages I build for AL2023 in COPR using ELEP9 do not always build (for various reasons).
Look into tighter integration with OpenELA and it's released packages.

**Comment by mimmus** [comment]
> Every time a package is requested to be included to AL2023 the community is asking Amazon to support it too. 
> Every time. Red Hat does not even do that for RHEL.

OK but with what timing?

What we need, before adopting AL2023 as base OS for the whole company (+150 accounts, +500 EC2), is to understand if we can rely on a packages repo at least as rich as EPEL.
If not, when we need to install some application having specific dependencies, this could became a problem.

**Reply by danie-dejager** [reply]
You can currently rebuild most packages from EPEL in COPR using a AL2023 chroot. I've done so here:
https://copr.fedorainfracloud.org/coprs/faramirza/al2023/
I keep it up-to-date myself. Ideally a community repo should exist, as with EPEL, to maintain the currency of the packages. There should not be too many packages you are not finding in AL2023. Missing packages should be seen as improving the Base OS (not because they are not the latest). I'd argue a missing package could be something like Fail2Ban but if you need the latest version of something PostgreSQL, above the AL2023 included versions, then you're on your own. Who would support you if you install PostgreSQL on AL2023 and something goes wrong? The developers would al least require versions for the BuildDependecies before it could effectively help. If AL2023 do not have a current Developer Toolset that could become a problem. I've not run into many scenarios where AL2023 was not able to rebuild Centos Stream or Fedora packages. Amazon provides the base OS and supports those packages. The ask is simply too big to do a blanket rebuild of the whole EPEL for AL2023.

**Comment by mimmus** [comment]
Developers pushes for Almalinux but we are worried because its 'legal' status could be easily change, as Red Hat/IBM is very aggressive on this field
